platform/windows/os_windows.cpp

```diff
@@ -1850,16 +1850,26 @@ String OS_Windows::get_cache_path() const {
 		if (has_environment("LOCALAPPDATA")) {
 			cache_path_cache = get_environment("LOCALAPPDATA").replace("\\", "/");
 		}
-		if (cache_path_cache.is_empty() && has_environment("TEMP")) {
-			cache_path_cache = get_environment("TEMP").replace("\\", "/");
-		}
 		if (cache_path_cache.is_empty()) {
-			cache_path_cache = get_config_path();
+			cache_path_cache = get_tmp_path();
 		}
 	}
 	return cache_path_cache;
 }
 
+String OS_Windows::get_tmp_path() const {
+	static String tmp_path_cache;
+	if (tmp_path_cache.is_empty()) {
+		if (tmp_path_cache.is_empty() && has_environment("TEMP")) {
+			tmp_path_cache = get_environment("TEMP").replace("\\", "/");
+		}
+		if (tmp_path_cache.is_empty()) {
+			tmp_path_cache = get_config_path();
+		}
+	}
+	return tmp_path_cache;
```

Comment:
Windows should probably use [GetTempPathW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettemppathw) instead of reading env. variable directly.